### PR TITLE
Wsbuffer

### PR DESCRIPTION
**Pull request by sticilface**
Remove debugging and finish constructors

### CODE REVIEW

File: src/AsyncEventSource.cpp [hunk 1]
```diff
@@ -115,19 +159,50 @@ AsyncEventSourceClient::AsyncEventSourceClient(AsyncWebServerRequest *request, A
...
   _client->onError(NULL, NULL);
-  _client->onAck(NULL, NULL);
-  _client->onPoll(NULL, NULL);
+  _client->onAck([](void *r, AsyncClient* c, size_t len, uint32_t time){ ((AsyncEventSourceClient*)(r))->_onAck(len, time); }, this);
+  _client->onPoll([](void *r, AsyncClient* c){ ((AsyncEventSourceClient*)(r))->_onPoll(); }, this);
   _client->onData(NULL, NULL);
-  _client->onTimeout([](void *r, AsyncClient* c __attribute__((unused)), uint32_t time){ ((AsyncEventSourceClient*)(r))->_onTimeout(time); }, this);
-  _client->onDisconnect([](void *r, AsyncClient* c){ ((AsyncEventSourceClient*)(r))->_onDisconnect(); delete c; }, this);
+  _client->onTimeout([this](void *r, AsyncClient* c __attribute__((unused)), uint32_t time){ ((AsyncEventSourceClient*)(r))->_onTimeout(time); }, this);
+  _client->onDisconnect([this](void *r, AsyncClient* c){ ((AsyncEventSourceClient*)(r))->_onDisconnect(); delete c; }, this);
```
Comment (me-no-dev) [inline src/AsyncEventSource.cpp]: Is [this] really needed?

**Comment by me-no-dev** [comment]
Is this otherwise good? What are Multi and nonMulti messages? Difference? Example?

**Comment by sticilface** [comment]
A multi message is one that is sent using '_ALL'  i.e. it is destined for multiple clients.  This way it is retained until all the clients have the message.  A non multi message is only destined for one client and can therefore be deleted once it has been sent.  


Adding some examples would be beneficial.. i agree.. 

Oh.. crap.. I think this pull request has all my other ones in it... darn!

**Comment by sticilface** [comment]
WSbuffer should just be the websocketsc stuff and not the eventsource stuff.

**Comment by sticilface** [comment]
So, i accidentally merged in that for some reason. 
The others should be separate.  
I'm making a fresh PR for the websocketsc stuff.
Give me an hour or so.  sorry i didn't notice that merge has snuck in there

**Pull request closed without merging**